### PR TITLE
Generative Classifier (main)

### PR DESCRIPTION
**Pull request by sheldon-roberts**
This pull request introduces `GenerativeClassifier`, a new model that treats text classification as a sequence‑generation task.  Instead of attaching a classification head on top of a transformer embedding, we fine‑tune a causal language model to generate the ground‑truth label string.

Key Benefits:
1. **Supports Extreme Classification** - The model can output an arbitrary number of labels without changing the final layer size.
2. **Zero/Few‑Shot Performance** - Leverage prompting with the pre-trained model to achieve strong few-shot performance.
3. **Constrained Decoding** – A prefix‑trie keeps generation strictly inside the allowed label set.
4. **Semantic label reasoning** – Label names are natural text, letting the model reason about their meaning and similarity.

## Usage example
```python
from transformers import AutoModelForCausalLM, AutoTokenizer
from flair.datasets import GO_EMOTIONS
from flair.trainers import ModelTrainer
from flair.data import Sentence
from flair.models import GenerativeClassifier

corpus = GO_EMOTIONS().downsample(0.01)
LABEL_TYPE = "emotion"
MODEL_NAME = "Qwen/Qwen2.5-0.5B"

label_dict = corpus.make_label_dictionary(label_type=LABEL_TYPE, add_unk=False)
tok = AutoTokenizer.from_pretrained(MODEL_NAME)
lm = AutoModelForCausalLM.from_pretrained(MODEL_NAME)

classifier = GenerativeClassifier(
    causal_model=lm,
    tokenizer=tok,
    label_dictionary=label_dict,
    label_type=LABEL_TYPE,
    prompt_template=("<Task> Classify the emotion of the following text </Task> Text: {text}  \nEmotions: "),
)

trainer = ModelTrainer(classifier, corpus)
trainer.fine_tune(
    base_path="resources/taggers/gencls_goemotions",
    mini_batch_size=8,
    max_epochs=5,
)

s = Sentence("I cant wait to try it! It looks so good")
classifier.predict(s)
print(s)
```

### CODE REVIEW

**Review by jeffpicard — APPROVED** [review]
LGTM! (this is a duplicate of a previous [PR](https://github.com/ZipRecruiter/flair/pull/13), but to a different branch).
